### DISCUSSION TITLE
`SubscriberExt::with` cant be used on a Box<dyn Subscriber>.

### DISCUSSION THREAD

**Discussion by Devlyn-Nelson** [tokio-rs/tracing#2722] [Q&A]
Is there a reason `tracing_subscriber::sealed::Sealed` is private? because it removes the ability to use `SubscriberExt` on something like a `Box<dyn Subscriber>` because i can't require `Sealed`.

**Comment by hawkw** [comment]
`SubscriberExt` is implemented for all `S` where `S` implements `Subscriber`. The trait is sealed to prevent it from being implemented for types that don't implement `Subscriber`, as this wouldn't be useful or make sense. 

If you can't call `with` on a `Box<dyn Subscriber>`, that's because `dyn Subscriber` doesn't implement `Subscriber`. We should fix this by adding an `impl Subscriber for dyn Subscriber` in the `tracing-core` crate.

**Reply by Devlyn-Nelson** [reply]
are we talking about the same thing? https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.SubscriberExt.html

**Reply by hawkw** [reply]
Yes, note that there is a blanket implementation of `SubscriberExt` for all `S` where `S: Subscriber`: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.SubscriberExt.html#impl-SubscriberExt-for-S

**Reply by Devlyn-Nelson** [reply]
I see. I don't get it why its Subscriber + Sealed if it gets blanketed to any Subscriber anyway, but i will trust that it makes sense in the larger picture. I will comeback to this when i have more time to mess around with nonessential improvements. Thank you and all the other tokio devs.

**Reply by hawkw** [reply]
I don't think that sealing the trait to prevent downstream implementations was actually necessary, and we could definitely remove it. But, we would still need to solve the problem of a missing subscriber implementation for `Box<dyn Subscriber>`.
